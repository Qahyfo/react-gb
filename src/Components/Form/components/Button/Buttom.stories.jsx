import React from 'react';
import { Button } from './Button';

export default {
  argTypes: {
    click: { action: 'click' },
  },
  component: Button,
  title: 'MyComponents/Button',
};

const Template = (arg) => <Button {...arg} />;

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
  label: 'test',
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
  label: 'test',
};
