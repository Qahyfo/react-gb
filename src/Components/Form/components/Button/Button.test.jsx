import { Button } from './Button';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

describe('Button', () => {
  it('render component', () => {
    render(<Button label="test" />);
  });

  // it('render witch snapshot', () => {
  //   const { asFragment } = render(<Button label="test" />)

  //   expect(asFragment()).toMatchSnapshot();
  // })

  it('component render witch text', () => {
    render(<Button label="test" />);

    expect(screen.getByText(/test/)).toBeInTheDocument();
  });

  it('render multiply component', () => {
    render(
      <>
        <Button label="test1" />
        <Button label="test2" />
      </>
    );

    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disabled', () => {
    render(<Button label="test" disabled />);

    expect(screen.getByText(/test/)).toBeDisabled();
  });

  // it('button have style color red', () => {
  //   render(<Button label="test" />);

  //   expect(screen.getByText(/test/)).toHaveStyle({ color: 'red' });
  // });

  it('button click witch userEvent', async () => {
    const mockHangler = jest.fn();
    render(<Button label="test" click={mockHangler} />);

    await userEvent.click(screen.getByText(/test/));

    expect(mockHangler).toHaveBeenCalledTimes(1);
  });

  it('button click async', async () => {
    const mockHangler = jest.fn();
    render(<Button label="test" click={() => setTimeout(mockHangler, 500)} />);

    await userEvent.click(screen.getByText(/test/));

    waitFor(() => expect(mockHangler).toHaveBeenCalledTimes(1));
  });

  it("test example")
});
