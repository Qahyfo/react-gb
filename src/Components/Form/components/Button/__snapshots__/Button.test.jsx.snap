// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render witch snapshot 1`] = `
<DocumentFragment>
  <button>
    test
  </button>
</DocumentFragment>
`;
